packages/components/pages/f-mfa/src/components/_tests/Mfa.test.js

```diff
@@ -80,36 +106,180 @@ describe('Mfa', () => {
+
+            // Act
+            await wrapper.setData({ showValidationError });
+
+            // Assert
+            const validationError = wrapper.find('[data-test-id="mfa-form-validation-error"]');
+            expect(validationError.exists()).toBe(showValidationError);
+        });
+
+        describe('And the form data is valid', () => {
+            it('should call the postValidateMfaToken() method with the correct params', async () => {
+                // Arrange
+                mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
+                const expectedParams = { mfa_token: 'XYZ1234', otp: 'test-otp' }; // eslint-disable-line camelcase
+                wrapper = await mountSut();
+                await wrapper.setData({ otp: 'test-otp', mfa: 'XYZ1234' });
+
+                // Act
+                await wrapper.vm.onFormSubmit();
+
+                // Assert
+                expect(mockPostValidateMfaToken).toHaveBeenCalledWith(expectedParams);
+            });
+
+            describe('And the submitting of the form data was unsuccessful', () => {
+                it.each([
+                    ['Bad request when submitting MFA', 400],
+                    ['Throttled when submitting MFA', 429]
+                ])("should log the error '%s' if the response status is '%s'", async (logMessage, status) => {
+                    // Arrange
+                    const errLogged = new Error('some status error message');
+                    errLogged.response = { status };
+                    mockPostValidateMfaToken = jest.fn(() => { throw errLogged; }); // Add error mock to the spy
+                    wrapper = await mountSut();
+                    await wrapper.setData({ otp: 'test-otp' });
+
+                    // Act
+                    await wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(errorLogSpy).toHaveBeenCalledWith(logMessage, errLogged, ['account-pages', 'mfa']);
+                    expect(wrapper.vm.isSubmitting).toBe(false);
+                });
+            });
+
+            describe('And the submitting of the form data was successful', () => {
+                it('should emit correct event if api call successful', async () => {
+                    // Arrange
+                    mockPostValidateMfaToken = jest.fn(() => {}); // Add default mock to the spy
+                    const expectedUrl = '/place/i/came/from';
+                    wrapper = await mountSut();
+                    await wrapper.setData({ otp: 'test-otp' });
+
+                    // Act
+                    await wrapper.vm.onFormSubmit();
+
+                    expect(wrapper.emitted()[REDIRECT_URL_EVENT_NAME]).toBeTruthy();
+                    expect(wrapper.emitted()[REDIRECT_URL_EVENT_NAME].length).toEqual(1);
+                    expect(wrapper.emitted()[REDIRECT_URL_EVENT_NAME][0]).toEqual([expectedUrl]);
```

Comment:
I think you only need this one. The others should be covered in this one expect (apart from the `.toEqual(1)` not sure how important that is?)